### ISSUE TITLE
Room setting constantly lost

### ISSUE THREAD

**Issue by addisonjames** [brian-su/homebridge-sonos-so-simple#31]
**Describe The Bug:**
After choosing a custom Room for my Beam device, the room is lost and resets to the default Room. This has been happening for months/as long as I've used the plugin. 

**To Reproduce:**
I'm not sure how long it takes to reset to default room, but it is definitely =<24 hours.

**Expected behavior:**
The Room setting is retained forever just like all other devices.

**Environment:**

* **Plugin Version**: v0.3.0 (2023-01-02)
* **Homebridge Version**: v4.50.2
* **Node.js Version**: [v16.16.0](https://homebridge.io/w/JTKEF)
* **NPM Version**: ?
* **Operating System**: Raspbian

Love the plugin, thanks in advance!

**Comment by brian-su** [comment]
That's an interesting one. Just so I fully understand, is this in the Home app that the Beam is resetting back to whatever room it has at first setup? Or is this in the Homebridge UI that it's resetting back to "default room"?

**Comment by addisonjames** [comment]
Yes the first example you gave- in iOS Home.app

**Comment by brian-su** [comment]
Ok, I'll have a look! 🙂

**Comment by brian-su** [comment]
So I've been trying to recreate this issue but can't seem to be able to. Whatever I try the device stays in the same room I assign it. 

![image](https://user-images.githubusercontent.com/13001723/213024272-45141379-d729-4efb-a8e0-8d4650ad5712.jpeg)

I'm really stumped by this one. Could you please paste is the config json you have for plugin? I'm not sure if that'll help but there could be a setting I don't have turned on that's messing it up.

**Comment by addisonjames** [comment]
Yes, i'm wondering if this was only an issue before iOS 16.0. Since I last posted here, my Beam has stayed in the custom room I set it to.

If you want to close this, I can let you know if it happens again!

Appreciate you!

**Comment by addisonjames** [comment]
Hey, it actually happened again. I'm not sure when exactly. The Beam moved from my "Living Room" to the room "Utility". No idea why it went to Utility, that's the last room in my list per my custom order.
I did have a power outage a couple days ago, but no idea if it had changed rooms before or after that, or if that even matters.
<img width="494" alt="CleanShot 2023-02-27 at 13 54 28@2x" src="https://user-images.githubusercontent.com/4740129/221694614-e96d0446-5158-413c-a295-d76a327a9571.png">

**Comment by brian-su** [comment]
Hmmm ok. I'll open the issue and take another wee look!

**Comment by addisonjames** [comment]
Happened again, I took this screenshot 6 days ago. But nowtoday I can't actually find the beam anywhere in my Home. I did update my HB to v1.6.1 this morning, but I'm not sure if it disappeared before or after that.
<img width="906" alt="CleanShot 2023-05-15 at 17 27 49@2x" src="https://github.com/brian-su/homebridge-sonos-so-simple/assets/4740129/4aede9aa-ba93-4429-95cd-fe8c5d827478">

**Comment by addisonjames** [comment]
I realize this is probably not very helpful, but it happened again. This time within ~2 weeks. The only thing I've done in HB in the last couple weeks is update other plugins and installed 1 new plugin.

If there's something I can do to monitor the change to help you out let me know. 

![image](https://github.com/brian-su/homebridge-sonos-so-simple/assets/4740129/12a661a8-8fcb-451e-8408-6400623995d8)

**Comment by brian-su** [comment]
Really bizarre. I still have yet to see that happen. I'll have another crack at trying to replicate it at the weekend!

**Comment by brian-su** [comment]
Lost room setting for me on 0.5.1. But no idea why. Will have another dig at the weekend

**Comment by j30542** [comment]
Hopefully this may clue everyone in on what could be causing this behavior. I am finding that when rooms are grouped, if homebridge/sonos plugin restarts, it will no longer acknowledge the soundbars and purge them. Immediately after ungrouping rooms, the soundbar accessories for speech and night will reappear on the following plugin reboot in a default room.

**Comment by addisonjames** [comment]
Hi, it's been a while- the Beam now stays put in my desired room 👍

Another thing I noticed is I moved the 'Hub' for the plugin to to the same. room. Not sure when I did that or if it's related to the fix or not. 

Running So Simple v0.5.4 and HB v1.11.0.

issue is closed as far as i'm concerned!  🙂

**Comment by brian-su** [comment]
Thanks for getting back in touch! Glad to hear the issue has been resolved.
